src/ParticleDistributions/ParticleDistributions.jl

```diff
@@ -467,7 +603,37 @@ function check_moment_consistency(m::Array{FT}) where {FT<:Real}
 
   nothing
 end
+# TODO: merge this functionality with update_params
+"""
+    update_dist_from_moments!(pdist::GammaPrimitiveParticleDistribution{FT}, moments::Array{FT})
+
+Updates parameters of the gamma distribution given the first three moments
+"""
+function update_dist_from_moments!(pdist::GammaPrimitiveParticleDistribution{FT}, moments::Array{FT}; param_range = Dict("θ" => (eps(FT), Inf), "k" => (eps(FT), Inf))) where {FT<:Real}
+  if length(moments) != 3
+    throw(ArgumentError("must specify exactly 3 moments for gamma distribution"))
+  end
+  if moments[1] > eps(FT)
```

Comment:
Why do you only check the first moment? It is possible that other moments are negative (maybe by small values due to ODE solver stuff); then the dist parameters will be negative too and the program will not continue. I suggest we check for all of the moments and only update if all of them are well defined.